tunix/rl/agentic/agents/tool_agent.py

```diff
@@ -0,0 +1,156 @@
+import copy
+import json
+import logging
+import uuid
+from typing import Any
+
+from tunix.rl.agentic.agents.base_agent import LLMBaseAgent, Step, Action, Trajectory
+from tunix.rl.agentic.tools.base_tool import BaseTool
+from tunix.rl.agentic.tools.tool_manager import ToolManager
+from tunix.rl.agentic.parser.tool_parser.tool_parser_registry import get_tool_parser
+from tunix.rl.agentic.parser.tool_parser.tool_parser_base import ToolParser
+
+logger = logging.getLogger(__name__)
+
+
+class ToolAgent(LLMBaseAgent):
+    """
+    An Agent implementation that supports tool usage, conforming to the LLMBaseAgent abstract interface.
+    Supports injecting tool_map and parsing with a specified parser.
+    """
+
+    def __init__(
+        self,
+        system_prompt: str,
+        parser_name: str = "qwen",
+        tool_map: dict[str, type[BaseTool]] | None = None,
+    ):
```

Comment:
add pydoc, esp for tool_map, what's the key and what's the value, is key the tool name or tool description?